showcase/src/elm/UrlGenerator.elm

```diff
@@ -0,0 +1,513 @@
+module UrlGenerator exposing (fromSourceCode)
+
+import Parser exposing ((|.), (|=), DeadEnd, Parser, Problem(..), Step(..))
+
+
+
+-- Can also be put into Utils
+-- but it might get pretty complicated so it's probably better to be put in its own file.
+
+
+fromSourceCode : String -> String -> String
+fromSourceCode version elmCode =
+    let
+        { title, code } =
+            elliefy elmCode
+
+        titleUrl =
+            encodeUrl title
+
+        elmCodeUrl =
+            encodeUrl code
+
+        htmlCodeUrl =
+            encodeUrl htmlCode
+
+        -- Package parsing is interesting because every package has its own '&packages='
+        packagesUrl =
+            packages version
+                |> List.map (packageToString >> encodeUrl)
+                |> List.map (\p -> "&packages=" ++ p)
+                |> String.join ""
+
+        link =
+            "https://ellie-app.com/a/example/v1?title="
+                ++ titleUrl
+                ++ "&elmcode="
+                ++ elmCodeUrl
+                ++ "&htmlcode="
+                ++ htmlCodeUrl
+                ++ packagesUrl
+                ++ "&elmversion=0.19.0"
+    in
+    link
+
+
+
+---- HTML
+
+
+htmlCode : String
+htmlCode =
+    """<html>
+<head>
+  <style>
+    /* you can style your program here */
+  </style>
+</head>
+<body>
+  <main></main>
+  <script>
+    var app = Elm.Main.init({ node: document.querySelector('main') })
+    // you can use ports and stuff here
+  </script>
+</body>
+</html>
+"""
+
+
+
+---- PACKAGES
+
+
+type alias Package =
+    { name : String
+    , version : String
+    }
+
+
+
+-- Elm-antd will have dynamic versioning
+
+
+packages : String -> List Package
+packages version =
+    [ Package "elm/browser" "1.0.2"
+    , Package "elm/core" "1.0.5"
+    , Package "elm/html" "1.0.0"
+    , Package "supermacro/elm-antd" version
+    ]
+
+
+packageToString : Package -> String
+packageToString p =
+    p.name ++ "@" ++ p.version
+
+
+
+---- ENCODING
+-- replacing every reserved URL character using (most likely) the most inefficient way possible
+-- reserved characters (from Wikipedia): ! * ' ( ) ; : @ & = + $ , / ? % # [ ]
+
+
+encodeUrl : String -> String
+encodeUrl =
```

Comment:
Renaming to `encodeUrlComponent` definitely makes it better. 

I actually never thought of the `elm/url` package for encoding! I took a look, and it seems like it will be a pretty clean solution. I implemented it in one of the commits and it works pretty well. Thanks!